contributors/design-proposals/node/node-usernamespace-remapping.md

```diff
@@ -48,134 +49,286 @@ Enable user namespace support in a kubernetes cluster so that workloads that wor
+    // The kubelet's runtime manager will set this to CONTAINER explicitly for v1 pods.
+    // Namespaces currently set by the kubelet: POD, CONTAINER, NODE
+    NamespaceMode pid = 2;
+    // IPC namespace for this container/sandbox.
+    // Note: There is currently no way to set CONTAINER scoped IPC in the Kubernetes API.
+    // Namespaces currently set by the kubelet: POD, NODE
+    NamespaceMode ipc = 3;
+    // User namespace for this container/sandbox
+    // Note: There is currently no way to set CONTAINER scoped user namespace in the Kubernetes API.
+    // The container runtime should IGNORE this if user namespace remapping is NOT supported.
+    // Kubelet will set it to POD if the following is true:
+    // - pod spec field `HostUserNamespace` == nil OR false
+    // - AND host usernamespace requiring configuration like hostpath volumes not there in pod
+    // - AND userns remapping is enabled at runtime  
+    // Kubelet will set it to NODE if the following is true:
+    // - userns remapping is NOT enabled at runtime
+    // - OR remapping enabled at runtime AND pod spec field `HostUserNamespace` == false
+    // - OR remapping enabled at runtime AND host usernamespace requiring configuration like hostpath volumes present in pod definition
+    // Namespaces currently set by the kubelet: NODE, POD
+    NamespaceMode user = 4;
 }
-message GetRuntimeConfig {
-    LinuxUserNamespaceConfig user_namespace_config = 1;
+
+// A NamespaceMode describes the intended namespace configuration for each
+// of the namespaces (Network, PID, IPC, User) in NamespaceOption. Runtimes should
+// map these modes as appropriate for the technology underlying the runtime.
+enum NamespaceMode {
+    // A POD namespace is common to all containers in a pod.
+    // For example, a container with a PID namespace of POD expects to view
+    // all of the processes in all of the containers in the pod.
+    // The uid/gids of the pods/containers on a given node are mapped to a range of uids/gids
+    // from that Node's namespace. For e.g. starting with 200000 id on the nodes namespace,
+    // 10000 ids are allocated for a pod. This means uid/gid 0 inside the pod would map to 200000 id on the nodes namespace.
+    // For example, starting from uid/gid 0, 10000 uids/gids in this namespace are mapped to
+    // 10000 ids on node namespace,starting with id 200000. i.e uid/gid 0 in this namespace is
+    // mapped to uid/gid 200000 on node namespace.
+    POD       = 0;
+    // A CONTAINER namespace is restricted to a single container.
+    // For example, a container with a PID namespace of CONTAINER expects to
+    // view only the processes in that container.
+    CONTAINER = 1; 
+    // A NODE namespace is the namespace of the Kubernetes node.
+    // For example, a container with a PID namespace of NODE expects to view
+    // all of the processes on the host running the kubelet.
+    NODE      = 2; 
```

Comment:
which of these will be default ? What is the current mode when i am not using any of these features ?